src/Compilers/CSharp/Portable/Binder/Binder.cs

```diff
@@ -773,6 +773,19 @@ internal Binder WithPatternVariablesIfAny(ArgumentListSyntax initializerArgument
             return new PatternVariableBinder(initializerArgumentListOpt, initializerArgumentListOpt.Arguments, this);
         }
 
+        internal void BuildAndAddPatternVariables(
+            ArrayBuilder<LocalSymbol> builder,
+            CSharpSyntaxNode node = null,
+            ImmutableArray<CSharpSyntaxNode> nodes = default(ImmutableArray<CSharpSyntaxNode>))
+        {
+            var patterns = PatternVariableFinder.FindPatternVariables(node, nodes);
+            foreach (var pattern in patterns)
+            {
+                builder.Add(SourceLocalSymbol.MakeLocal(ContainingMemberOrLambda, this, RefKind.None, pattern.Type, pattern.Identifier, LocalDeclarationKind.PatternVariable));
+            }
+            patterns.Free();
```

Comment:
This is a good suggestion, however, since this change targets stabilization, I prefer to scope amount of changes to a necessary minimum that addresses the problem. I will open an issue to make sure we clean this up in the feature branch.